packages/platform/test/IndexedDbDatabase.test.ts

```diff
@@ -0,0 +1,232 @@
+import {
+  IndexedDb,
+  IndexedDbDatabase,
+  IndexedDbMigration,
+  IndexedDbQuery,
+  IndexedDbTable,
+  IndexedDbVersion
+} from "@effect/platform"
+import { afterEach, assert, describe, it } from "@effect/vitest"
+import { Effect, Layer, Schema } from "effect"
+import { IDBKeyRange, indexedDB } from "fake-indexeddb"
+
+const databaseName = "db"
+
+const layerFakeIndexedDb = Layer.succeed(IndexedDb.IndexedDb, IndexedDb.make({ indexedDB, IDBKeyRange }))
+
+const provideMigration = (migration: IndexedDbMigration.IndexedDbMigration.Any) =>
+  Effect.provide(
+    IndexedDbQuery.layer.pipe(
+      Layer.provide(
+        IndexedDbDatabase.layer(databaseName, migration).pipe(Layer.provide(layerFakeIndexedDb))
+      )
+    )
+  )
+
+afterEach(() => {
+  indexedDB.deleteDatabase(databaseName)
+})
+
+describe("IndexedDbDatabase", () => {
+  it.effect("insert and read todos", () => {
+    const Table = IndexedDbTable.make(
+      "todo",
+      Schema.Struct({
+        id: Schema.Number,
+        title: Schema.String,
+        completed: Schema.Boolean
+      }),
+      { keyPath: "id", indexes: { titleIndex: "title" } }
+    )
+
+    const Db = IndexedDbVersion.make(Table)
+
+    class Migration extends IndexedDbMigration.make(Db, (api) =>
+      Effect.gen(function*() {
+        yield* api.createObjectStore("todo")
+        yield* api.createIndex("todo", "titleIndex")
+        yield* api.from("todo").insert({ id: 1, title: "test", completed: false })
+      }))
+    {}
+
+    return Effect.gen(function*() {
+      const { makeApi, use } = yield* IndexedDbQuery.IndexedDbApi
+      const api = makeApi(Db)
```

Comment:
This doesn't feel quite right yet. Maybe a field on the IndexedDbVersion would be better.

Reply:
I moved `IndexedDbApi` inside `IndexedDbVersion`. Do you think is possible to include the api directly inside the `IndexedDbVersion` interface? Not sure because it's a context with a generic parameter

Reply:
Something like:

```ts
const api = yield* Db
// or
const api = yield* Db.api
```

Reply:
Done